internal/zinc-persist/src/main/scala/sbt/internal/inc/FileAnalysisStore.scala

```diff
@@ -45,6 +45,8 @@ object FileAnalysisStore {
 
     private final val format = new BinaryAnalysisFormat(readWriteMappers)
     private final val TmpEnding = ".tmp"
+    private final val useCompression =
+      sys.props.get("zinc.binary-file-store.use-compression").map(_ == "true").getOrElse(true)
```

Comment:
Should we add this flag to any of the configuration objects of the incremental compiler? Why does this exist? Do we expect to gain some performance improvement by not compressing in some situations?

Reply:
For local builds when we don't worry about the space it is just faster to store this without compression and also then read it without decompressing. Though it is rather unrelated to this PR and I don't have any numbers on it and I don't want to spend time on measuring that. I can just revert this change and keep it in private for to maybe introduce it later. Or if it is fine like this we can keep it.

Reply:
Either you expose a field to modify the compression in Zinc's public API or you remove it completely, I'm fine with any. But let's not leave it as is.

Reply:
I am not sure how to add it in an elegant way, so for now I remove it, to not spend too much time. I think I may come back to this later on a separate PR.

Reply:
Sounds good to me. The more we do to unblock this PR, the better.